src/tribler-core/tribler_core/modules/metadata_store/community/gigachannel_community.py

```diff
@@ -1,342 +1,238 @@
-            )
-
-    @lazy_wrapper(SearchResponsePayload)
-    async def on_search_response(self, peer, response):
-        search_request_cache = self.request_cache.get(u"remote-search-request", response.id)
-        if not search_request_cache or not search_request_cache.process_peer_response(peer):
-            return
+    def legacy_send_remote_select(self, peer, processing_callback=None, **kwargs):
 
-        def _process_received_blob():
-            md_results = self._update_db_with_blob(response.raw_blob)
-            if not md_results:
-                self.metadata_store.disconnect_thread()
-                return None, None
-
-            # it is incorrect to call md.simple_dict() for metadata objects from md_results
-            # as previous db_session is already over, so we are going to fetch them again in a new db_session
-            md_ids = [
-                md.rowid for md, action in md_results
-                if md
-                   and (md.metadata_type in [CHANNEL_TORRENT, REGULAR_TORRENT])
-                   and action in [UNKNOWN_CHANNEL, UNKNOWN_TORRENT, UPDATED_OUR_VERSION, UNKNOWN_COLLECTION]
-            ]
+        request = SelectRequest(self.request_cache, hexlify(peer.mid), kwargs, processing_callback)
+        self.request_cache.add(request)
 
-            with db_session:
-                md_list = self.metadata_store.ChannelNode.select(lambda md: md.rowid in md_ids)[:]
-                result = (
-                    [md.to_simple_dict() for md in md_list],
-                    gen_have_newer_results_blob(md_results),
-                )
-            self.metadata_store.disconnect_thread()
-            return result
+        self.logger.info(f"Select to {hexlify(peer.mid)} with ({kwargs})")
+        self.ez_send(peer, LegacyRemoteSelectPayload(request.number, json.dumps(kwargs).encode('utf8')))
 
-        search_results, reply_blob = await get_event_loop().run_in_executor(None, _process_received_blob)
+    @lazy_wrapper(LegacySelectResponsePayload)
+    async def legacy_on_remote_select_response(self, peer, response_payload):
+        """
+        Match the the response that we received from the network to a query cache
+        and process it by adding the corresponding entries to the MetadataStore database.
+        This processes both direct responses and pushback (updates) responses
+        """
+        self.logger.info(f"Legacy response from {hexlify(peer.mid)}")
 
-        if self.notifier and search_results:
-            self.notifier.notify(
-                NTFY.REMOTE_QUERY_RESULTS, {"uuid": search_request_cache.uuid, "results": search_results}
-            )
+        # ACHTUNG! the returned request cache can be either a SelectRequest or PushbackWindow
+        request = self.request_cache.get(hexlify(peer.mid), response_payload.id)
+        if request is None:
+            self.logger.info(f"No request for response")
```

Comment:
No need for an f-string here.

Reply:
+1 from pylint_check :) https://jenkins-ci.tribler.org/job/GH_Tribler_PR_Tests/job/PR_pylint_check/1420//artifact/output/quality_report.html

Reply:
Fixed